src/pkg/packager/sources/tarball.go

```diff
@@ -0,0 +1,300 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package sources contains core implementations of the PackageSource interface.
+package sources
+
+import (
+	"archive/tar"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/oci"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/mholt/archiver/v3"
+)
+
+var (
+	// veryify that TarballSource implements PackageSource
+	_ PackageSource = (*TarballSource)(nil)
+
+	// veryify that SplitTarballSource implements PackageSource
+	_ PackageSource = (*SplitTarballSource)(nil)
+)
+
+// TarballSource is a package source for tarballs.
+type TarballSource struct {
+	*types.ZarfPackageOptions
+}
+
+// LoadPackage loads a package from a tarball.
+func (s *TarballSource) LoadPackage(dst *layout.PackagePaths) (err error) {
+	var pkg types.ZarfPackage
+
+	message.Debugf("Loading package from %q", s.PackageSource)
+
+	if s.Shasum != "" {
+		if err := utils.SHAsMatch(s.PackageSource, s.Shasum); err != nil {
+			return err
+		}
+	}
+
+	pathsExtracted := []string{}
+
+	err = archiver.Walk(s.PackageSource, func(f archiver.File) error {
```

Comment:
```suggestion
	// Walk the package so that was can dynamically load a .tar or a .tar.zst without caring about filenames.
	err = archiver.Walk(s.PackageSource, func(f archiver.File) error {
```